test-framework/test-suites/unit/tests/pylib/stack/test_pylib_stack_util.py

```diff
@@ -95,3 +95,9 @@ def test_unique_everseen(self, key, test_input, expected):
 	def test_lowered(self):
 		"""Expect lowered to lowercase all strings in an interable."""
 		assert ["foo", "bar", "baz"] == list(stack.util.lowered(["FOO", "BAR", "BAZ"]))
+
+	def test_is_valid_hostname(self):
+		assert stack.util.is_valid_hostname("backend-0-0")
+		assert not stack.util.is_valid_hostname("-no-leading-minus")
+		assert not stack.util.is_valid_hostname("no-trailing-minus-")
+		assert not stack.util.is_valid_hostname("no.other_chars")
```

Comment:
Maybe add q couple of the edge cases for the regex pattern? Host with one/zero letters, host that's too long, etc.

Reply:
Incorporated in [d723ee1](https://github.com/Teradata/stacki/pull/641/commits/d723ee184381189d251f39ebd548fcf18a4bebac).